src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DataBaseHandler.java

```diff
@@ -0,0 +1,233 @@
+            throws SQLException, ConnectException {
+        String sqlInsertStatement = String.format("INSERT INTO %s(%s,%s) VALUES(?,?)", table, EVENT_ID_KEY,
+                table.keyName);
+        executeUpdate(sqlInsertStatement, keyValue, eiffelevent);
+
+    }
+
+    /**
+     * This function will create a database structure in the database file if it does not exist.
+     *
+     * @throws ConnectException
+     */
+    private void createNewDatabase() throws ConnectException {
+        try (Connection connection = connect()) {
+            if (connection != null) {
+                connection.getMetaData();
+                connection.close();
+            }
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to create new database: {}\n{}", e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This function loops the values in the Table enum and executes the create table command for each
+     * table and creates it in the database if it not already exist.
+     *
+     * @throws ConnectException
+     */
+    private void createTables() throws ConnectException {
+        try (Connection connection = connect(); Statement statement = connection.createStatement()) {
+            for (Table table : Table.values()) {
+
+                // Add a check if table is ALL, then add extra fields
+                String sqlCreateStatement = String.format(
+                        "CREATE TABLE IF NOT EXISTS %s (%s text PRIMARY KEY, %s text)", table,
+                        table.keyName,
+                        EVENT_ID_KEY);
+                statement.execute(sqlCreateStatement);
+            }
+
+            LOGGER.debug("Created tables successfully");
+        } catch (SQLException e) {
+            LOGGER.error("Error while creating Tables in database: {}\n{}", e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This function connects to the database and returns a connection
+     *
+     * @return Connection to database
+     * @throws ConnectException
+     */
+    private Connection connect() throws ConnectException {
+        try {
+            Class.forName("org.sqlite.JDBC");
+            return DriverManager.getConnection(this.databaseFile);
+        } catch (SQLException | ClassNotFoundException e) {
+            throw new ConnectException(
+                    String.format("Failed to create a database connection. %s\n%s", e.getMessage(), e));
+        }
+    }
+
+    /**
+     * executes the prepared statement.
+     *
+     * @param preparedStatement
+     * @return
+     * @throws SQLException
+     */
+    private String executeQuery(final PreparedStatement preparedStatement) throws SQLException {
+        try (ResultSet result = preparedStatement.executeQuery();) {
+            if (result.next()) {
+                return result.getString(EVENT_ID_KEY);
+            }
+        } catch (SQLException e) {
+            throw e;
```

Comment:
I guess it should log before it throws again?

Reply:
Logging is made in the catcher above, this try/catch simply just ensure the preparedStatement is properly closed. Logging is different depending on who calls this method.

Reply:
Why not just skip the try/catch?

Reply:
See that now... You can skip the catch block as `SQLException` is in the throws declaration.

Reply:
Yea, If you don't log the error then there is no need for the try catch block.

Reply:
done